### DISCUSSION TITLE
Definition of the ligand group

### DISCUSSION THREAD

**Discussion by iwatobipen** [dptech-corp/Uni-Dock#123] [Q&A]
Dear developper,
I hope everything goes well.
I have question about the definition of the ligand group in unidock.
The group of ligand such as small, medium, large and extra large is defined the main.cpp the URL is below.
https://github.com/dptech-corp/Uni-Dock/blob/ab4d8cd46886949da10e62a5f10cd5d8839ef76f/unidock/src/main/main.cpp#L1020
The code divides into quarter the ligand list without considering properties of lignads suchas num of rotatable bonds, num of atoms etc....
Is it suitable way to define group of each ligand?
I asked this topic because I got cuda runtime error when I run the unidock and the code defined ligands with large num_lig_pairs as medium or smallGroup.
I would like to know how to fix the error.
Any advice or suggestions will be greatly appreciated

Thanks,
Taka

**Comment by ysyecust** [comment]
Hi @iwatobipen, Thank you for your reply, at present, the grouping in our project takes into account the number of atoms, the number of rotatable keys and other factors, and the grouping is carried out after sorting, the detailed code can be found in this link https://github.com/dptech-corp/Uni-Dock/blob/ab4d8cd46886949da10e62a5f10cd5d8839ef76f/unidock/src/main/main.cpp#L1016, if you are convenient, you can talk about your more detailed error message submission, and we will further improve the sorting mechanism in the future.

**Reply by iwatobipen** [reply]
Hi @ysyecust, Thanks for your promt reply.
I tried to run docking study and screening test example the repository provided finished witout any error. 
I used docker image which dptech team provided.
However when I used different dataset, I got following error.
num_lig_pair of the bach is 478 but it's assigined as mediumGroup but following code defiened maximum valule is 330.
https://github.com/dptech-corp/Uni-Dock/blob/ab4d8cd46886949da10e62a5f10cd5d8839ef76f/unidock/src/cuda/kernel.h#L146
So I'm wondering the miss much assignment causes error.


****snip****
   2       -9.689       1.48      3.469
   3       -9.597      1.441      1.839
   4       -9.481     0.9402      1.419

mode |   affinity | dist from best mode
     | (kcal/mol) | rmsd l.b.| rmsd u.b.
-----+------------+----------+----------
   1       -7.936          0          0
   2       -7.783      2.762      6.372
poses saveing time: 19284
Batch 2 running time: 1233361ms
Medium
Max num_atoms: 42 Max num_torsions: 12 Max num_rigids: 6 Max num_lig_pairs: 478
Group size: 8860
Medium Batch 3 size: 2392
WARNING: At low exhaustiveness, it may be impossible to utilize all CPUs.
Performing docking (random seed: -2041814946) ... CUDA error at /opt/unidock/src/cuda/monte_carlo.cu:2854 code=700(cudaErrorIllegalAddress) "cudaDeviceSynchronize()"


An error occurred: CUDA Runtime Error.

Please report bugs through the Issue Tracker on GitHub
(https://github.com/dptech-corp/Uni-Dock/issues)., so
that this problem can be resolved. The reproducibility of the
error may be vital, so please remember to include the following in your problem report:
* the EXACT error message,
* your version of the program,
* the type of computer system you are running it on,
* all command line options,
* configuration file (if used),
* ligand file as PDBQT,
* receptor file as PDBQT,
* flexible side chains file as PDBQT (if used),
* output file as PDBQT (if any),
* input (if possible),
* random seed the program used (this is printed when the program starts).

Thank you!

Thanks,
@iwatobipen
